plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMySqlTypeMapping.java

```diff
@@ -129,7 +118,112 @@ public void testBoolean()
                 .addRoundTrip("boolean", "false", TINYINT, "TINYINT '0'")
                 .addRoundTrip("boolean", "NULL", TINYINT, "CAST(NULL AS TINYINT)")
                 .execute(getQueryRunner(), mysqlCreateAndInsert("tpch.test_boolean"))
-                .execute(getQueryRunner(), trinoCreateAsSelect("tpch.test_boolean"));
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_boolean"))
+                .execute(getQueryRunner(), trinoCreateAndInsert("test_boolean"));
```

Comment:
Writes via trino will map the boolean to a bit instead of tinyint.

Something like this instead.

```java
SqlDataTypeTest.create()
    .addRoundTrip("boolean", "true", BOOLEAN)
    .addRoundTrip("boolean", "false", BOOLEAN)
    .addRoundTrip("boolean", "NULL", BOOLEAN, "CAST(NULL AS BOOLEAN)")
    .execute(getQueryRunner(), trinoCreateAsSelect("test_boolean"))
    .execute(getQueryRunner(), trinoCreateAndInsert("test_boolean"));
```

Reply:
Sorry for the delay. I was just listening to a sharing and just finished.

The conclusion I observed is that if we write `BOOLEAN` type through Trino, it will be mapped as `tinyint(1)` in MySQL (we can refer to [this code](https://github.com/trinodb/trino/blob/master/plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClient.java#L400-L402), if you look at this code, you may have a question, didn't we already map Trino's `BOOLEAN` to MySQL's `boolean` when we wrote it? Haha, the reason is that MySQL `boolean` is actually MySQL `tinyint(1)`, you can refer to [this doc](https://dev.mysql.com/doc/refman/8.0/en/numeric-type-syntax.html)), so if we read it through Trino, it will be mapped to Trino's `TINYINT` as well.

IMO, we should fix it this way:

```
        SqlDataTypeTest.create()
                .addRoundTrip("boolean", "true", TINYINT, "TINYINT '1'")
                .addRoundTrip("boolean", "false", TINYINT, "TINYINT '0'")
                .addRoundTrip("boolean", "NULL", TINYINT, "CAST(NULL AS TINYINT)")
                .execute(getQueryRunner(), mysqlCreateAndInsert("tpch.test_boolean"))
                .execute(getQueryRunner(), trinoCreateAsSelect("test_boolean"));

        SqlDataTypeTest.create()
                .addRoundTrip("boolean", "1", TINYINT, "TINYINT '1'")
                .addRoundTrip("boolean", "0", TINYINT, "TINYINT '0'")
                .addRoundTrip("boolean", "NULL", TINYINT, "CAST(NULL AS TINYINT)")
                .execute(getQueryRunner(), trinoCreateAndInsert("test_boolean"));
```

Reply:
I fixed it as I said in the comments above.